raiden/network/transport/matrix/transport.py

```diff
@@ -1099,17 +1100,9 @@ def _get_public_room(self, room_name: str, invitees: List[User]) -> Room:
                 )
             else:
                 self.log.debug("Room created successfully", room=room, invitees=invitees)
-                break
+                return room
         else:
-            # if can't join nor create, create an unnamed one
-            room = self._client.create_room(None, invitees=invitees_uids, is_public=True)
-            self.log.warning(
-                "Could not create nor join a named room. Successfuly created an unnamed one",
-                room=room,
-                invitees=invitees,
-            )
-
-        return room
+            raise error
```

Comment:
I don't really see the point of this `else` really. This would be executed in case the for loop doesn't loop on anything, which is not the case provided that we loop over `range(JOIN_RETRIES)`. 

Also, even if we endup in the else block, the error would be `None` so `raise None` would result in:
```
>>> raise None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: exceptions must derive from BaseException
```

Reply:
@rakanalh actually, `else` after `for` and `while` loops, in python, are run iff the loop didn't `break` nor `return`ed. It's useful in constructs like `try N times, if succeed break, else continue; else didnt break, re-raise something`, so it can be hit even looping over `range(JOIN_RETRIES)`, as long as no iteration succeeded/broke/returned.

Reply:
TIL... thanks @andrevmatos

Reply:
Ok here's an experiment that Augusto and I put together to make sure the change makes sense:
```
In [1]: def f():
   ...:     e = None
   ...:     try:
   ...:         raise Exception('h')
   ...:     except Exception as e:
   ...:         pass
   ...:     return e
   ...:

In [2]: f()
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-2-0ec059b9bfe1> in <module>()
----> 1 f()

<ipython-input-1-1e88e880d26a> in f()
      5     except Exception as e:
      6         pass
----> 7     return e
      8

UnboundLocalError: local variable 'e' referenced before assignment
```

This means that 
```
except MatrixRequestError as error:
```
Would just shadow the original `error` variable and won't set it. As a result, the original `error` would always be `None`.

That being said, i think this PR would be merged **iff** a test was written to simulate this raise.